hoomd/md/external/wall.py

```diff
@@ -0,0 +1,577 @@
+"""
+
+from hoomd.md import force
+from hoomd.data.array_view import _ArrayViewWrapper
+from hoomd.md import _md
+import hoomd
+
+
+def _to_md_cpp_wall(wall):
+    if isinstance(wall, hoomd.wall.Sphere):
+        return _md.SphereWall(wall.radius, wall.origin.to_base(), wall.inside,
+                              wall.open)
+    if isinstance(wall, hoomd.wall.Cylinder):
+        return _md.CylinderWall(wall.radius, wall.origin.to_base(),
+                                wall.axis.to_base(), wall.inside, wall.open)
+    if isinstance(wall, hoomd.wall.Plane):
+        return _md.PlaneWall(wall.origin.to_base(), wall.normal.to_base(),
+                             wall.open)
+    raise TypeError(f"Unknown wall type encountered {type(wall)}.")
+
+
+class _WallArrayViewFactory:
+
+    def __init__(self, cpp_wall_potential, wall_type):
+        self.cpp_obj = cpp_wall_potential
+        self.func_name = {
+            hoomd.wall.Sphere: "get_sphere_list",
+            hoomd.wall.Cylinder: "get_cylinder_list",
+            hoomd.wall.Plane: "get_plane_list"
+        }[wall_type]
+
+    def __call__(self):
+        return getattr(self.cpp_obj.field, self.func_name)()
+
+
+class WallPotential(force.Force):
+    r"""Generic wall potential.
+
+    Warning:
+        `WallPotential` should not be used directly.  It is a base class that
+        provides features and documentation common to all standard wall
+        potentials.
+
+    All wall potential classes specify that a given potential energy and
+    force be computed on all particles in the system when the signed cutoff
+    distance is near the wall surface: :math:`r < r_{\mathrm{cut}}`.
+    The force :math:`\vec{F}` is in the direction of :math:`\vec{r}`, the vector
+    pointing from the particle to closest point on the wall's surface and
+    :math:`V_{\mathrm{pair}}(r)` is the pair potential specified by subclasses
+    of `WallPotential`.  Walls are two-sided surfaces with positive signed
+    distances to points on the active side of the wall and negative signed
+    distances to points on the inactive side. Additionally, the wall's mode
+    controls how forces and energies are computed for particles on or near the
+    inactive side. The ``inside`` flag (or ``normal`` in the case of
+    `hoomd.wall.Plane`) determines which side of the surface is active.
+
+    .. rubric:: Standard Mode.
+
+    In the standard mode, when :math:`r_{\mathrm{extrap}} \le 0`, the potential
+    energy is only computed on the active side. :math:`V(r)` is evaluated in the
+    same manner as when the mode is shift for the analogous :py:mod:`pair
+    <hoomd.md.pair>` potentials within the boundaries of the active space.
+
+    .. math::
+
+        V(r)  = V_{\mathrm{pair}}(r) - V_{\mathrm{pair}}(r_{\mathrm{cut}})
+
+    For ``open=True`` spaces:
+
+    .. math::
+        :nowrap:
+
+        \begin{eqnarray*}
+        \vec{F}  = & -\nabla V(r) & 0 < r < r_{\mathrm{cut}} \\
```

Comment:
```suggestion
        \vec{F}  = & -\nabla V(r) \,\,\,& 0 < r < r_{\mathrm{cut}} \\
```

Reply:
What is this supposed to do? It doesn't seem to change anything in the rendered output.

Reply:
It adds a little bit of white space between the functional forms and valid domains. It was all mashed together and just looked like `V(r)0` without those spaces.